code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushNotificationBuilder.java

```diff
@@ -0,0 +1,502 @@
+ */
+class AEPPushNotificationBuilder {
+
+    private static final String SELF_TAG = "AEPPushNotificationBuilder";
+    private static final String DEFAULT_CHANNEL_ID = "CampaignPushChannel";
+    // When no channel name is received from the push notification, this default channel name is used.
+    // This will appear in the notification settings for the app.
+    private static final String DEFAULT_CHANNEL_NAME = "Campaign Classic General Notifications";
+
+    /**
+     * Builds a notification for the provided {@code AEPPushPayload}.
+     *
+     * @param payload {@link AEPPushPayload} created from the received push notification
+     * @param context the application {@link Context}
+     * @return the notification
+     */
+    @Nullable
+    static Notification buildPushNotification(final AEPPushPayload payload,
+                                              final Context context) {
+        Notification notification;
+        final Map<String, String> messageData = payload.getMessageData();
+        final PushTemplateType pushTemplateType = PushTemplateType.fromString((Objects.requireNonNull(messageData.get(CampaignPushConstants.PushPayloadKeys.TEMPLATE_TYPE))));
+        try {
+            switch (pushTemplateType) {
+                case BASIC:
+                    final BasicPushTemplate basicPushTemplate = new BasicPushTemplate(messageData);
+                    notification = createBasicTemplatePushNotification(basicPushTemplate, context);
+                    break;
+//            case AUTO_CAROUSEL:
+//                final CarouselPushTemplate carouselPushTemplate = new CarouselPushTemplate(messageData);
+//                notification = createAutoCarouselTemplatePushNotification(payload, context);
+//                break;
+//            case MANUAL_CAROUSEL:
+//                final CarouselPushTemplate carouselPushTemplate = new CarouselPushTemplate(messageData);
+//                notification = createManualCarouselTemplatePushNotification(payload, context);
+//                break;
+//            case INPUT_BOX:
+//                final InputBoxPushTemplate inputBoxPushTemplate = new InputBoxPushTemplate(messageData);
+//                notification = createInputBoxTemplatePushNotification(payload, context);
+//                break;
+                case UNKNOWN:
+                case LEGACY:
+                default:
+                    final AEPPushTemplate aepPushTemplate = new AEPPushTemplate(messageData);
+                    notification = createLegacyPushNotification(aepPushTemplate, context);
+                    break;
+            }
+        } catch (final IllegalArgumentException exception) {
+            Log.error(LOG_TAG, SELF_TAG, "Failed to create a Notification from the provided AEPPushPayload, an illegal argument exception occurred: %s", exception.getLocalizedMessage());
+            return null;
+        }
```

Comment:
Do not try/catch the exception and return null here. Instead throw this all the way up ( to  `handleRemoteMessage ` where the exception is already handled for a different ). This will prevent from creating empty notifications and centralizing the handling for improper payloads.